### PR TITLE
allow set from json

### PR DESCRIPTION
**Pull request by benoist**

### CODE REVIEW

File: src/json/from_json.cr [hunk 1]
```diff
@@ -66,6 +66,20 @@ def Array.new(pull : JSON::PullParser)
   end
 end
 
+def Set.new(pull : JSON::PullParser)
+  ary = new
+  new(pull) do |element|
+    ary << element
+  end
+  ary
```
Comment (lbguilherme) [inline src/json/from_json.cr]: Não chamaria de `ary` aqui, afinal não é uma array

Comment (benoist) [inline src/json/from_json.cr]: Quer que eu mude o nome para Set?

File: src/json/from_json.cr [hunk 2]
```diff
@@ -66,6 +66,20 @@ def Array.new(pull : JSON::PullParser)
...
+end
+
+def Set.new(pull : JSON::PullParser)
+  pull.read_array do
+    yield T.new(pull)
+  end
+end
```
Comment (lbguilherme) [inline src/json/from_json.cr]: I'm not sure I agree with the existence of this method. It yields multiple times the same element, which doesn't make much sense for a `Set`. It also doesn't return a `Set`, which violates its name: `new`.

File: spec/std/json/mapping_spec.cr [hunk 3]
```diff
@@ -185,4 +189,11 @@ describe "JSON mapping" do
     json.key.should eq(1)
     json.pull.should eq(2)
   end
+
+  it "parses json array as set" do
+    json = JsonArrayAndSet.from_json(%({"array": ["a", "a", "b"], "set": ["a", "a", "b"]}))
+    json.array.should eq(["a", "a", "b"])
+    json.set.should eq(Set(String).new(["a", "b"]))
```
Comment (jhass) [inline spec/std/json/mapping_spec.cr]: The shorthand should work here, `Set(String) {"a", "b"}`

File: src/json/from_json.cr (hunk 2 above)
Comment (jhass) [inline src/json/from_json.cr]: Yeah, just reuse `Array.new(JSON::PullParser)`

**Comment by benoist** [comment]
comments processed and rebased

**Comment by asterite** [comment]
:+1:

**Comment by jhass** [comment]
Thanks!

**Pull request merged**